spacy/tokens/doc.pyx

```diff
@@ -285,6 +286,10 @@ cdef class Doc:
                     sent_starts[i] = -1
                 elif sent_starts[i] is None or sent_starts[i] not in [-1, 0, 1]:
                     sent_starts[i] = 0
+        if pos is not None:
+            for pp in pos:
```

Comment:
will it be quicker if we loop over `set(pos)` here?

Reply:
Ah, good point, that should have fewer `not in` checks.

Reply:
```suggestion
            for pp in set(pos):
```